test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/validators/AllSpansAreInOrder.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.test.telemetry.tracing.validators;
+
+import org.opensearch.test.telemetry.tracing.MockSpanData;
+import org.opensearch.test.telemetry.tracing.TracingValidator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * AllSpansAreInOrder validator to check if all spans are emitted in order.
+ */
+public class AllSpansAreInOrder implements TracingValidator {
```

Comment:
Let's remove this test for now. As this doesn't hold true after the changes made in this PR - https://github.com/opensearch-project/OpenSearch/pull/8831